pypika/terms.py

```diff
@@ -288,57 +288,111 @@ def get_sql(self, **kwargs: Any) -> str:
         raise NotImplementedError()
 
 
+def idx_placeholder_gen(idx: int) -> str:
+    return str(idx + 1)
+
+
+def named_placeholder_gen(idx: int) -> str:
+    return f'param{idx + 1}'
+
+
 class Parameter(Term):
     is_aggregate = None
 
     def __init__(self, placeholder: Union[str, int]) -> None:
         super().__init__()
-        self.placeholder = placeholder
+        self._placeholder = placeholder
```

Comment:
Might be good to use a setter here. With that, implementing a check for the correct type can happen on initialization if put in the setter

Reply:
I'm not sure I follow how to do this pythonically? Could you share some pseudo code to show what you mean?

Reply:
@mvanderlee I believe @wd60622 was referring to the following:
```python
@property
def placeholder(self):
      return self._placeholder

@placeholder.setter
def placeholder(self, placeholder):
     # can add checks here if needed
     self._placeholder = placeholder
```

Reply:
Right, which adds no value and isn't pythonic. This isn't Java/C#.

It would also break parameters. After a parameter has been created and used, any changes to the placeholder will break queries. Which is why I've made it _private_ with a getter only.

I haven't changed the type, so any type checks would have to be part of a separate PR.

Reply:
I would disagree with you on it not being Pythonic, as it’s a native language feature. 

Also, I was trying to be helpful, but given your receptiveness to discussion and tone, I won’t proceed. 

Good luck.

Reply:
Sorry for my tone @danielenricocahall it'd been a rough week and I lashed out.

As for pythonic. This video does a good job at explaining language feature vs pythonic. 
Raymond Hettinger - Beyond PEP 8 -- Best practices for beautiful intelligible code - https://www.youtube.com/watch?v=wf-BqAjZb8M